Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.h

```diff
@@ -191,6 +192,11 @@ namespace EMotionFX
             ActorAsset::ActorInstancePtr        m_actorInstance;            ///< Live actor instance.
             AZStd::unique_ptr<RenderActorInstance> m_renderActorInstance;
 
+            uint32_t m_lightingChannelMask = 1;
+            bool m_lightingChannel0 = true;
+            bool m_lightingChannel1 = false;
+            bool m_lightingChannel2 = false;
```

Comment:
This seems like a very useful feature. The biggest issue that I see is the way that the data is being represented in all of the components and controllers. Ideally, you would only need one variable stored and reflected per class to represent the lighting channel mask. I'm assuming that it's separated into blocks like this so that it's easier to reflect and edit in the edit context. This will make it extremely challenging if you want to add more than three lighting channels in the future.

My recommendation would be to change all of the variables in component based classes to use a single AZStd::array<bool, N> or AZStd::fixed_vector<bool, N>. N is the maximum number of lighting channels and should be defined globally. Then you only have to reflect one variable. The edit context will only need one entry and it should automatically create a checkbox for every element. Set up the data element attributes so that the container cannot be modified or reordered. Also make sure that it starts at a collapsed state so that it doesn't take up a lot of UI real estate by default.

IIRC, those containers have a specialization when the type is bool to store one element per bit.

I also recommend using uint32_t consistently. I see alternating uses of uint32_t and uint.

Reply:
Hi, I changed the code to gather the channels inside a class, so that we won't need to update everywhere once we want more lighting channels.

For using uint32_t consistently, this version use only uint32_t in C++ side, and use uint in the shader.